Composer/packages/client/src/recoilModel/dispatchers/lg.ts

```diff
@@ -208,40 +211,27 @@ export const lgDispatcher = () => {
       try {
         const { set, snapshot } = callbackHelpers;
         //set content first
-        set(lgFilesState(projectId), (prevLgFiles) => {
-          return prevLgFiles.map((file) => {
-            if (file.id === id) {
-              return {
-                ...file,
-                content,
-              };
-            }
-            return file;
-          });
+        set(lgFileState({ projectId, lgFileId: id }), (prevLgFile) => {
+          return {
+            ...prevLgFile,
+            content,
+          };
         });
 
-        const lgFiles = await snapshot.getPromise(lgFilesState(projectId));
+        const lgFiles = await snapshot.getPromise(lgFilesSelectorFamily(projectId));
         const updatedFile = (await LgWorker.parse(projectId, id, content, lgFiles)) as LgFile;
         const updatedFiles = await getRelatedLgFileChanges(projectId, lgFiles, updatedFile);
 
         // compare to drop expired change on current id lg file.
         /**
          * Why other methods do not need double check content?
-         * Because this method already did set content before call lgFilesAtomUpdater.
+         * Because this method already did set content before call updateLgFiles.
          */
-        set(
-          lgFilesState(projectId),
-          lgFilesAtomUpdater({ updates: updatedFiles }, (prevLgFiles) => {
-            const targetInState = prevLgFiles.find((file) => file.id === updatedFile.id);
-            const targetInCurrentChange = updatedFiles.find((file) => file.id === updatedFile.id);
-            // compare to drop expired content already setted above.
-            if (targetInState?.content !== targetInCurrentChange?.content) {
-              return (lgFile) => lgFile.id !== updatedFile.id;
-            } else {
-              return () => true;
-            }
-          })
-        );
+
+        updateLgFiles(callbackHelpers, projectId, { updates: updatedFiles }, (current, changed) => {
+          // compare to drop expired content already setted above.
+          return current?.content === changed?.content;
```

Comment:
We need also compare id, because "updateFiles" can be other lg file, and the content of course is different , then will always drop their changes. Fixed this in #5654